src/wallet/coinselection.cpp

```diff
@@ -168,6 +170,30 @@ bool SelectCoinsBnB(std::vector<OutputGroup>& utxo_pool, const CAmount& selectio
     return true;
 }
 
+std::optional<std::pair<std::set<CInputCoin>, CAmount>> SelectCoinsSRD(const std::vector<OutputGroup>& utxo_pool, CAmount target_value)
+{
+    std::set<CInputCoin> out_set;
+    CAmount value_ret = 0;
+
+    std::vector<size_t> indexes;
+    indexes.resize(utxo_pool.size());
+    std::iota(indexes.begin(), indexes.end(), 0);
+    Shuffle(indexes.begin(), indexes.end(), FastRandomContext());
```

Comment:
Wouldn't it be easier to shuffle the `utxo_pool` directly? Or is this the same object that is used for the other selection algorithms?

Reply:
This change was suggested in https://github.com/bitcoin/bitcoin/pull/17526#discussion_r625212261